src/colp/components/agency_usetype_report.py

```diff
@@ -0,0 +1,83 @@
+import streamlit as st
+from src.constants import COLOR_SCHEME
+from abc import ABC
+import plotly.express as px
+
+
+class CountRecordsReport(ABC):
+    def __call__(self):
+        if self.data is None:
+            st.info(f"No Count of records by {self.y_axis_col}")
+            return
+        self.set_title()
+        self.plot()
+
+    def set_title(self):
+        by_agency = ""
```

Comment:
It might be cleaner to move this function into the subclasses, so there are three set_title functions but each is basically just

```
def set_title(self):
  st.subheader(f"City owned and leased properties by agency")

def set_title(self):
  st.subheader(f"City owned and leased properties by usetype")

def set_title(self):
  st.subheader(f"City owned and leased properties by agency and usetype")
```
or even dryer,

in __call__
```
st.subheader(self.title())
```
where title just returns the string
and you forget the by_agency and by_usetype booleans.

Reply:
got it, I had fun setting the title programmatically but I agree it's not worth the length and non-readability